### PR TITLE
Modernize functional tests

### PR DESCRIPTION
**Pull request by mdellweg**
[noissue]

### CODE REVIEW

**Comment by mdellweg** [comment]
This is surely not all. But this commit is quite big already.

File: pulp_ansible/tests/functional/cli/test_collection_upload.py [hunk 1]
```diff
@@ -16,7 +17,7 @@
 )
 from pulp_smash.pulp3.bindings import delete_orphans, monitor_task, PulpTestCase
 from pulp_smash.pulp3.utils import gen_distribution, gen_repo
-from pulp_smash.utils import http_get, uuid4
+from pulp_smash.utils import http_get
```
Comment (gerrod3) [inline pulp_ansible/tests/functional/cli/test_collection_upload.py:20]: We have a fixture for `http_get`.

**Review by gerrod3 — COMMENTED** [review]
What pulp_smash is left?

File: pulp_ansible/tests/functional/api/collection/test_remote.py [hunk 2]
```diff
@@ -1,60 +1,59 @@
...
-    def test_update_auth_url(self):
-        """Assert that a `CollectionRemote` with `token` and no `auth_url` can be created."""
-        body = gen_ansible_remote(
-            url="https://example.com/",
-            token="this is a token string",
-            auth_url="https://example.com",
+@pytest.mark.parallel
+def test_auth_url_requires_token(ansible_collection_remote_factory):
+    """Assert that a `CollectionRemote` with `auth_url` and no `token` can't be created."""
+    with pytest.raises(ApiException):
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply pulp_ansible/tests/functional/api/collection/test_remote.py:43]: That's worth a try.

File: pulp_ansible/tests/functional/api/collection/test_remote.py [hunk 3]
```diff
@@ -1,60 +1,59 @@
...
 from pulpcore.client.pulp_ansible.exceptions import ApiException
-from pulp_smash.pulp3.bindings import monitor_task
 
-from pulp_ansible.tests.functional.utils import gen_ansible_remote, TestCaseUsingBindings
-from pulp_ansible.tests.functional.utils import set_up_module as setUpModule  # noqa:F401
 
+@pytest.mark.parallel
+def test_remote_with_url_only_is_allowed(ansible_collection_remote_factory):
+    """Assert that a `CollectionRemote` with only a url can be created."""
+    ansible_collection_remote_factory(url="https://example.com/")
```
[the remark this answers is not in the document]
Comment (mdellweg) [reply pulp_ansible/tests/functional/api/collection/test_remote.py:8]: assert on what? If it wouldn't be allowed, the factory shall raise an error.

**Pull request merged**